src/v/cluster/metadata_dissemination_service.cc

```diff
@@ -97,8 +97,12 @@ ss::future<> metadata_dissemination_service::start() {
                 return;
             }
             auto ntp = c->ntp();
-            handle_leadership_notification(
-              std::move(ntp), term, std::move(leader_id));
+            auto cs = _topics.local().materialized_children(ntp);
```

Comment:
wouldn't that be better to query source partition leadership instead ?

Reply:
Do you mean we should query which partition is the source partition for given ntp `ntp`? That's pretty much what this does, only returns the result set in ntps since the topic hierarchy lookup table is a `map<topic_namespace, topic_namespace>`. Does that address your concern?

Reply:
I mean we do not need to send materialized leader updates to the other nodes since it is derived from the source partition. Calling `handle_leadership_notification` for materialized topics dispatches RPC calls related with those topics. Information about source partition leadership is already disseminated. I mean it would be enough to handle `is_leader` in a way where materialized partition would query its source for leadership.

Reply:
I see your point. I did this because I wanted to make sure that every materialized ntp exists in the partition leaders table. This is desired because we want materialized topics to behave like normal ones. Why do we want it in this table? Because then if it isn't its considered a special case. And every time you want to look something up, you'll first need to ask, 'is this materialized?' Previously we knew if it wasn't or not when we had a topic naming schema `<source>.$<dest>$` but now that we've abandoned this I figure the next best thing is to just keep all relevant data structures filled with materialized ntps as if they were normal ones. 

I think a good balance might be to update the partition leaders table without making this RPC call. I will investigate